File: app/src/main/java/com/test/mercaridemo/network/NetworkModule.java
package com.test.shoppingapp.network;

import com.test.networkmodule.NetworkSDK;
import com.test.networkmodule.api.ShoppingAPI;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Singleton
@Module
public class NetworkModule {

    @Provides
    @Singleton
    public NetworkSDK provideNetworkSDK() {
        return NetworkSDK.getInstance();
    }

    @Provides
    @Singleton
    public ShoppingAPI provideShoppingAPI(NetworkSDK networkSDK) {
        return networkSDK.getShoppingAPI();
    }
}

File: app/src/main/java/com/test/shoppingapp/network/NetworkModule.kt
package com.test.shoppingapp.network

import com.test.networkmodule.NetworkSDK
import com.test.networkmodule.api.ShoppingAPI
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Singleton
@Module
class NetworkModule {

    @Provides
    @Singleton
    fun provideNetworkSDK(): NetworkSDK {
        return NetworkSDK.getInstance()
    }

    @Provides
    @Singleton
    fun provideShoppingAPI(networkSDK: NetworkSDK): ShoppingAPI {
        return networkSDK.shoppingAPI
    }
}